src/sentry/sentry_apps/external_requests/select_requester.py

```diff
@@ -41,66 +49,83 @@ class SelectRequester:
     dependent_data: str | None = field(default=None)
 
     def run(self) -> SelectRequesterResult:
+
         response: list[dict[str, str]] = []
         url = None
-        try:
-            url = self._build_url()
-            body = safe_urlread(
-                send_and_save_sentry_app_request(
-                    url,
-                    self.sentry_app,
-                    self.install.organization_id,
-                    "select_options.requested",
-                    headers=self._build_headers(),
-                )
-            )
 
-            response = json.loads(body)
-        except Exception as e:
-            extra = {
+        with SentryAppInteractionEvent(
+            operation_type=SentryAppInteractionType.EXTERNAL_REQUEST,
+            event_type=SentryAppEventType.SELECT_OPTIONS_REQUESTED,
+        ).capture() as lifecycle:
+            extras: dict[str, Any] = {
                 "sentry_app_slug": self.sentry_app.slug,
                 "install_uuid": self.install.uuid,
                 "project_slug": self.project_slug,
             }
+            lifecycle.add_extras(extras)
+
+            try:
+                url = self._build_url()
+                extras.update({"url": url})
```

Comment:
this will not add to the lifecycle since you are only updating the `extras` dictionary. you probably need to do `lifecycle.add_extra("url", url)`, unless you only want to add these extras when a failure/halt happens?